core/src/main/scala/frameless/InjectionEnum.scala

```diff
@@ -0,0 +1,54 @@
+package frameless
+
+import scala.reflect.{ClassTag, classTag}
+import shapeless._
+
+object InjectionEnum {
+  type InjectionEnum[A] = Injection[A, String]
+
+  def apply[A: InjectionEnum]: InjectionEnum[A] = implicitly[InjectionEnum[A]]
+
+  def instance[A](f: A => String, g: String => A): InjectionEnum[A] =
+    Injection(f, g)
+
+  implicit val cnilInjectionEnum: InjectionEnum[CNil] =
```

Comment:
I think it should be possible to avoid the `CNil` instance by having a "`CLast`" case, something like this:

```scala
implicit def clastInjectionEnum[H: ClassTag]
  (implicit gen: Generic.Aux[H, HNil]): InjectionEnum[H :+: CNil]
```